File: Maths/Geometry/Pythagorean_theorem/Pythagorean_theorem.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {

	double sides[3];
	cout << "Enter the sides of triangle: ";
	cin >> sides[0] >> sides[1] >> sides[2];

	sort(sides, sides + 3);

	if((sides[0] * sides[0]) + (sides[1] * sides[1]) == sides[2] * sides[2]) {

		cout << "True, it is a right angled triangle as it follows Pythagorean Theorem (a^2 + b^2 = c^2)" << endl;

	}

	else {
		cout << "False, it is not a right angled triangle as it does not follow the Pythagorean Theorem" << endl;
	}

	return 0;
}

File: Maths/Geometry/Pythagorean_theorem/Pythagorean_theorem.rb
#!/usr/bin/env ruby

puts "Enter the sides of the triangle:"
sides = gets.chomp(" ")
            .split(" ")
            .map(&:to_f)
            .sort

if sides.length != 3
  raise ArgumentError, "Please enter the lengths of the three sides of a triangle"
end

if sides[0]**2 + sides[1]**2 == sides[2]**2
  puts "According to Pythagorean Theorem, a^2 + b^2 = c^2, you have a right triangle"
else
  puts "According to Pythagorean Theorem, you do not have a right triangle"
end
